### PR TITLE
Update session interface to match api changes

### PR DESCRIPTION
**Pull request by AshGodfrey**
## Description

Matching Node changes: 
 https://github.com/workos/workos-node/pull/802

## Documentation

Does this require changes to the WorkOS Docs? E.g. the [API Reference](https://workos.com/docs/reference) or code snippets need updates.

```
[ x] Yes
```

If yes, link a related docs PR and add a docs maintainer as a reviewer. Their approval is required.

### CODE REVIEW

File: pkg/users/client.go [hunk 1]
```diff
@@ -134,32 +130,39 @@ type ListUsersOpts struct {
...
 
-type AddUserToOrganizationOpts struct {
-	User         string `json:"id"`
-	Organization string `json:"organization_id"`
+type Session struct {
+	ID                        string                     `json:"id"`
+	Token                     string                     `json:"token"`
+	CreatedAt                 string                     `json:"created_at"`
+	ExpiresAt                 string                     `json:"expires_at"`
+	AuthorizedOrganizations   []Organization             `json:"authorized_organizations"`
```
Comment (oliverzheng) [inline pkg/users/client.go:145]: There should be an `AuthorizedOrganization` struct, which has just 1 field `organization` within it, which has name and ID. It looks a little extra but it's by design.

Comment (AshGodfrey) [reply pkg/users/client.go:145]: Will make that update, thanks for confirming

File: pkg/users/client.go [hunk 2]
```diff
@@ -134,32 +130,42 @@ type ListUsersOpts struct {
...
+const (
+	GoogleOauth    SessionAuthenticationMethod = "GoogleOauth"
+	MagicAuth      SessionAuthenticationMethod = "MagicAuth"
+	MicrosoftOauth SessionAuthenticationMethod = "MicrosoftOauth"
+	Password       SessionAuthenticationMethod = "Password"
+)
+
+type UnauthorizedOrganizationReason struct {
+	Type                         string                        `json:"type"`
+	AllowedAuthenticationMethods []SessionAuthenticationMethod `json:"allowedAuthenticationMethods"`
```
Comment (jonatascastro12) [inline pkg/users/client.go:163]:
```suggestion
	AllowedAuthenticationMethods []SessionAuthenticationMethod `json:"allowed_authentication_methods"`
```

**Review by jonatascastro12 — CHANGES_REQUESTED** [review]
Need to change that camelCase value to snake case since it reflects the json field.

**Review by jonatascastro12 — APPROVED** [review]

**Pull request merged**